### PR TITLE
add function to decrease on increase vote count

### PR DESCRIPTION
**Pull request by Kayphaz007**

### CODE REVIEW

File: src/components/ArticleCard.jsx [hunk 1]
```diff
@@ -13,6 +15,34 @@ const ArticleCard = ({
...
+    }
+  }
+  async function decreaseVote() {
+    try {
+      setVoteCount((prevVote) => {
+        return prevVote - 1;
+      });
+      await decreaseArticleVote(article_id);
+    } catch (err) {
+      console.log(err);
```
Comment (MKD01) [inline src/components/ArticleCard.jsx]: Console.log

File: src/utils/api.js [hunk 2]
```diff
@@ -20,3 +20,25 @@ export async function getArticleById(article_id) {
...
 }
+
+export async function increaseArticleVote(article_id) {
+  try {
+    const result = await baseApi.patch(`/articles/${article_id}`, {
+      inc_votes: 1,
+    });
+    return result;
+  } catch (err) {
+    console.log(err);
```
Comment (MKD01) [inline src/utils/api.js]: console.log

File: src/utils/api.js [hunk 3]
```diff
@@ -20,3 +20,25 @@ export async function getArticleById(article_id) {
...
+  }
+}
+export async function decreaseArticleVote(article_id) {
+  try {
+    const result = await baseApi.patch(`/articles/${article_id}`, {
+      inc_votes: -1,
+    });
+    return result;
+  } catch (err) {
+    console.log(err);
```
Comment (MKD01) [inline src/utils/api.js]: Do you need this here?

**Review by MKD01 — CHANGES_REQUESTED** [review]
Great work overall, everything seems to work as it should on the all articles page, however the ticket does specify:

Users should be able to vote on the article that they are reading.

Currently your logic all exists on the page with all articles, but when clicking within a single article to read more about it I am not able to vote on it. Users should also be able to vote on an article in this page as well. 

You also have a few extra console.log's that can be removed. 

Once you have added the above behaviour please re-submit this pr 👍

File: src/components/ArticleCard.jsx (hunk 1 above)
Comment (Kayphaz007) [reply src/components/ArticleCard.jsx]: I was told console.log for errors are allowed

File: src/components/Vote.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    try {
+      setVoteCount((prevVote) => {
+        return prevVote + 1;
+      });
+      await increaseArticleVote(article_id);
+    } catch (err) {
+      setVoteCount(() => {
+        return votes;
+      });
+      window.alert("There was an error");
```
Comment (popatre) [inline src/components/Vote.jsx:17]: is there a reason the window alert only happens for the upvote one?

File: src/components/Vote.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      setVoteCount((prevVote) => {
+        return prevVote - 1;
+      });
+      await decreaseArticleVote(article_id);
+    } catch (err) {
+      setVoteCount(() => {
+        return votes;
+      });
+    }
+  }
```
Comment (popatre) [inline src/components/Vote.jsx]: At a later date, could probably refactor these into one function. They're the same thing, except one increase it by 1, the other -1

File: src/utils/api.js [hunk 6]
```diff
@@ -5,18 +5,27 @@ const baseApi = axios.create({
...
-    console.log(err);
+    return Promise.reject("Error");
   }
 }
+export async function decreaseArticleVote(article_id) {
+  const result = await baseApi.patch(`/articles/${article_id}`, {
+    inc_votes: -1,
+  });
+  return result;
+}
```
Comment (popatre) [inline src/utils/api.js]: Id refactor these into one function where you just pass in 1 or -1 : but up to you.

**Review by popatre — APPROVED** [review]
- Good clear feedback that the vote count has changed
- Good to see the db being updated with the vote increment
- Excellent job handling the votes optimistically - creates a better and quicker UX/UI

- [ ] I've left a comment regarding refactoring your two functions that are basically the same, but ultimately up to you if you want to do it
- [ ]  - also, the window alert only shows for upvote rejections, not down votes. But again, up to you if you want to sort that

**Pull request merged**